src/pages/subject-detail-page/subject-detail-page.tsx

```diff
@@ -38,73 +38,108 @@ interface SubjectDetailPageProps extends VariantProps<typeof SubjectDetailPageVa
   const { subjectCode } = useParams()
-  const [ mainSubject, setMainSubject ] = useState<string>('Disciplina')
-  const [ subjects, setSubjects ] = useState(Array<Subject>)
-  const [ recurrency, setRecurrency ] = useState<LessonRecurrencyInterface>()
-  const [ lessonDatetimes, setLessonDatetimes ] = useState<LessonRecurrentDatetime[]>()
-  // const [ studentClass, setStudentClass ] = useState(Array<StudentClass>)
-  // const [ selected, setSelected ] = useState<number>(-1)
-  const [ selected, setSelected ] = useState<Subject>()
-  const [ ClassSelected, setClassSelected ] = useState<number>(1)
+  const mainSubject = MainSubject.find(ms => ms.id == subjectCode)!.name
+  const toastify = useToastify()
 
-  const [ isCreateOpen, setIsCreateOpen ] = useState(false)
-  const [ isEditOpen, setIsEditOpen ] = useState(false)
-  const [ isDeleteOpen, setIsDeleteOpen ] = useState(false)
-  const [ isAnimationEnded, setIsAnimationEnded ] = useState(true)
-  const [ isModalConfirm, setIsModalConfirm ] = useState(true)
+  const [ isModalOpen, setIsModalOpen ] = useState(false);
+  const [ modalType, setModalType ] = useState<'create' | 'update' | 'delete'>('create')
+  const openModal = (type: 'create' | 'update' | 'delete') => {
+    setModalType(type)
+    setIsModalOpen(true)
+  }
 
+  const [ subjects, setSubjects ] = useState(Array<Subject>)
+  const [ subjectIndex, setSubjectIndex ] = useState(0)
   useEffect(() => {
-    subjectCode && isModalConfirm && setMainSubject(MainSubject.find(ms => ms.id == subjectCode)!.name)
-
-    subjectCode && isModalConfirm &&
+    subjectCode &&
     Services.listSubjectsFromMain(subjectCode)
       .then(data => {
         setSubjects(data)
-        setIsModalConfirm(false)
+        setSubjectIndex((prev) => prev >= data.length ? 0 : prev)
       })
       .catch(error => {
         console.log(error)
       })
+  }, [!isModalOpen]);
 
-    subjectCode && selected &&
-    Services.listRecurrencyFromParams(selected.id, ClassSelected)
-      .then(data => {
-        console.log(data)
-        setRecurrency(data)
-        console.log(data)
-        setLessonDatetimes(data.lessonDatetimes)
-      })
-      .catch(error => {
-        setRecurrency(undefined)
-        setLessonDatetimes(undefined)
-        console.log(error)
-      })
+  const hendleSubjectIndex = (subjectId: number) => {
+    const index = subjects.findIndex((sub) => sub.id === subjectId)
+    setSubjectIndex(index)
+  }
 
-  }, [subjectCode, isModalConfirm, selected]);
+  const [ studentClasses, setStudentClasses ] = useState(Array<StudentClass>);
+  const [ classIndex, setClassIndex ] = useState(0);
+  useEffect(() => {
+    Services.listStudentClasses()
+      .then((data) => {
+        setStudentClasses(data);
+        setClassIndex((prev) => prev >= data.length ? 0 : prev)
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  }, [!isModalOpen]);
```

Comment:
Me corrija se eu tiver errado, mas aqui ele não ativa sempre que isModalOpen muda? 
Pq se for o caso, pra fazer a validação, mete um if isModalOpen return; dentro do useEffect e paz

Mas tu manja mais de front do que eu então foi uma dúvida msm